### ISSUE TITLE
When call to get-dependency fails, the reported component is always ::not-found [0.3.0]

### ISSUE THREAD

**Issue by hlship** [stuartsierra/component#40]
```
clojure.lang.ExceptionInfo: Missing dependency :redis-customer-db of clojure.lang.Keyword expected in system at :redis-customer-db
         component: :com.stuartsierra.component/not-found
    dependency-key: :redis-customer-db
            reason: :com.stuartsierra.component/missing-dependency
            system: {:tiny-url {:endpoint "http://wmt.co"},
                     :customer-receipts-service
                     {:port 8081,

                     ...}
        system-key: #object[clojure.core$key 0x4c075147 "clojure.core$key@4c075147"]
=> nil
```

This would be more useful if the component for whom the dependency is being resolved was reported; this is not the case:

```
(defn- get-dependency [system system-key component dependency-key]
  (let [component (get system system-key ::not-found)]
    (when (nil? component)
      (throw (nil-component system system-key)))
    (when (= ::not-found component)
      (throw (ex-info (str "Missing dependency " dependency-key
                           " of " (platform/type-name component)
                           " expected in system at " system-key)
                      {:reason ::missing-dependency
                       :system-key key
                       :dependency-key dependency-key
                       :component component
                       :system system})))
    component))
```

I believe the intent was to build the ex-info data and message around the component which was to receive the dependency, but the `(let [component ...`) masks the `component` parameter.  Refactoring the local 'component' symbol to `dependency` would help.

I'll put together a PR shortly.

**Comment by stuartsierra** [comment]
Thanks for the report; I will fix.

**Comment by hlship** [comment]
PR almost ready.

**Comment by stuartsierra** [comment]
fix included in release 0.3.1
